pyfifotax/data_structures_dataframe.py

```diff
@@ -0,0 +1,299 @@
+from datetime import datetime
+from dataclasses import dataclass, asdict
+import pandas as pd
+
+
+@dataclass
+class DataFrameRow:
+    def to_dict(self):
+        return asdict(self)
+
+    @staticmethod
+    def from_schwab_json(json_dict):
+        raise NotImplementedError
+
+    @staticmethod
+    def empty():
+        raise NotImplementedError
+
+    @staticmethod
+    def from_df_row(row):
+        raise NotImplementedError
+
+
+@dataclass
+class ESPPRow(DataFrameRow):
+    date: datetime
+    symbol: str
+    buy_price: pd.Float64Dtype
+    fair_market_value: pd.Float64Dtype
+    quantity: pd.Float64Dtype
+    currency: str
+
+    @staticmethod
+    def from_schwab_json(json_dict):
+        symbol = json_dict["Symbol"]
+        quantity = pd.to_numeric(json_dict["Quantity"])
+        if not len(json_dict["TransactionDetails"]) == 1:
+            raise RuntimeError(
+                "Could not convert ESPP information from Schwab JSON, expected TransactionDetails to be of length 1."
+            )
+        details = json_dict["TransactionDetails"][0]["Details"]
+        date = datetime.strptime(details["PurchaseDate"], "%m/%d/%Y")
+        buy_price = pd.to_numeric(details["PurchasePrice"].strip("$").replace(",", ""))
+        fair_market_value = pd.to_numeric(
+            details["PurchaseFairMarketValue"].strip("$").replace(",", "")
+        )
+        return ESPPRow(
+            date,
+            symbol,
+            buy_price,
+            fair_market_value,
+            quantity,
+            "USD",
+        )
+
+    @staticmethod
+    def empty():
+        return ESPPRow(None, None, None, None, None, None)
```

Comment:
Technically, valid point. 
The issue I have with default values is that you then indeed could use the constructor and potentially forget some positional arguments with default values. Does this make sense? :) I know, not really the best style.

Reply:
In this case, the solution is simple: `date: datetime|None`, without specifying the default :). This way neither the IDE is complaining, nor you have a default value.

Reply:
Ah, it's just about typing, sure :)

Reply:
changed it, the only use was to get the variable names of this dataclass in an empty dict, to it always was called with `empty().to_dict()` anyways, tried to avoid the None completely now by just setting dummy values within an `empty_dict()` method